.github/workflows/deploy.yml

```diff
@@ -14,7 +14,7 @@ on:
 
 jobs:
   deploy:
-    if: contains(github.event.head_commit.message, '[stagingdeploy]') || contains(github.event.head_commit.message, '[testnetdeploy]') || contains(github.event.head_commit.message, '[mainnetdeploy]')
+    if: github.event_name == 'release' || github.ref == 'refs/heads/main' || contains(github.event.head_commit.message, '[stagingdeploy]') || contains(github.event.head_commit.message, '[testnetdeploy]') || contains(github.event.head_commit.message, '[mainnetdeploy]')
```

Comment:
Some weeks ago, we removed this patch for safety, but this should exist.
This action should be run when the commit has the expected substrings, but also when:
- A PR gets merged to main, to deploy things to the **testnet** network.
- When a release event happens, so the **mainnet** job is run.